write-fonts/src/tables/glyf.rs

```diff
@@ -0,0 +1,298 @@
+                0 => (same_or_pos, CoordDelta::Skip),
+                SHORT_MIN..=-1 => (short_flag, CoordDelta::Short(value.unsigned_abs() as u8)),
+                1..=SHORT_MAX => (short_flag | same_or_pos, CoordDelta::Short(value as _)),
+                _other => (SimpleGlyphFlags::empty(), CoordDelta::Long(value)),
+            }
+        }
+
+        let (mut last_x, mut last_y) = (0, 0);
+        let mut iter = self.contours.iter().flatten();
+        std::iter::from_fn(move || {
+            let point = iter.next()?;
+            let mut flag = SimpleGlyphFlags::empty();
+            let d_x = point.x - last_x;
+            let d_y = point.y - last_y;
+            last_x = point.x;
+            last_y = point.y;
+
+            if point.on_curve {
+                flag |= SimpleGlyphFlags::ON_CURVE_POINT;
+            }
+            let (x_flag, x_data) = flag_and_delta(
+                d_x,
+                SimpleGlyphFlags::X_SHORT_VECTOR,
+                SimpleGlyphFlags::X_IS_SAME_OR_POSITIVE_X_SHORT_VECTOR,
+            );
+            let (y_flag, y_data) = flag_and_delta(
+                d_y,
+                SimpleGlyphFlags::Y_SHORT_VECTOR,
+                SimpleGlyphFlags::Y_IS_SAME_OR_POSITIVE_Y_SHORT_VECTOR,
+            );
+
+            flag |= x_flag | y_flag;
+            Some((flag, x_data, y_data))
+        })
+    }
+}
+
+/// A little helper for managing how we're representing a given delta
+#[derive(Clone, Copy, Debug)]
+enum CoordDelta {
+    // this is a repeat (set in the flag) and so we write nothing
+    Skip,
+    Short(u8),
+    Long(i16),
+}
+
+impl FontWrite for CoordDelta {
+    fn write_into(&self, writer: &mut crate::TableWriter) {
+        match self {
+            CoordDelta::Skip => (),
+            CoordDelta::Short(val) => val.write_into(writer),
+            CoordDelta::Long(val) => val.write_into(writer),
+        }
+    }
+}
+
+impl<'a> IntoIterator for &'a Contour {
+    type Item = &'a CurvePoint;
+
+    type IntoIter = std::slice::Iter<'a, CurvePoint>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.iter()
+    }
+}
+
+impl FontWrite for SimpleGlyf {
+    fn write_into(&self, writer: &mut crate::TableWriter) {
+        assert!(self.contours.len() < i16::MAX as usize);
+        assert!(self._instructions.len() < u16::MAX as usize);
+        let n_contours = self.contours.len() as i16;
+        n_contours.write_into(writer);
+        let bbox = [self.x_max, self.y_max, self.x_min, self.y_min];
+        bbox.write_into(writer);
+        // now write end points of contours:
+        let mut cur = 0;
+        for contour in &self.contours {
+            cur += contour.len();
+            (cur as u16 - 1).write_into(writer);
+        }
+        (self._instructions.len() as u16).write_into(writer);
+        self._instructions.write_into(writer);
+
+        let deltas = self.compute_point_deltas().collect::<Vec<_>>();
+        //TODO: calculate flag repeats here
```

Comment:
Suggest filing an issue unless you are going to more or less immediately address

Reply:
yes, there is a follow-up PR coming.